### PR TITLE
added kicks

### PR DESCRIPTION
**Pull request by pavelGotbit**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Kickspad

##### Twitter Link:
https://twitter.com/KicksPad_io

##### List of audit links if any:
NO

##### Website Link:
https://kickspad.io

##### Logo (High resolution, will be shown with rounded borders):
https://drive.google.com/file/d/1WSuUNXg4Pg1QZ-_NqOSiz7-ClyIWUi71/view

##### Current TVL:
https://bscscan.com/address/0x7f435dc15cd9d686c48b6bed2cb8d32d895cfb93

##### Treasury Addresses (if the protocol has treasury)
https://bscscan.com/address/0x7f435dc15cd9d686c48b6bed2cb8d32d895cfb93

##### Chain:
BNB Chain

##### Short Description (to be shown on DefiLlama):
Description: KicksPad is an IDO Launchpad, focusing on bringing the most outstanding projects in the web3 market and directly connecting them with people willing to invest. Our vision is to create transparent, rewarding, and riskless IDO investment opportunities for investors and a successful fundraise model for new web3 projects. The investment process will be conducted without any staking or KYC barriers and will have an additional staking reward system and a possibility of refunding the IDO investments. 

##### Token address and ticker if any:
0xfEB4e9B932eF708c498Cc997ABe51D0EE39300cf
$KICKS 

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Launchpad

##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):
NO

##### Implementation Details: Briefly describe how the oracle is integrated into your project:
NO

##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:
NO

##### forkedFrom (Does your project originate from another project):
NO

##### methodology (what is being counted as tvl, how is tvl being calculated):
TVL counted from Kicks contracts (including staking)

##### Github org/user (Optional, if your code is open source, we can track activity):
NO

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/kicks exports TVL: 
        
 
 
    bsc-staking               17.09 k
    staking                   17.09 k
    bsc                       0
    
    total                    0

File: projects/kicks/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const STAKING = "0x7f435dC15CD9d686C48B6beD2cB8d32D895cfb93";
+const TOKEN = "0xfEB4e9B932eF708c498Cc997ABe51D0EE39300cf";
+
+async function staking(time, ethBlock, _b, { api }) {
+  const supply = await api.call({
+    abi: "function totalSupply() external view returns (uint256)",
```
Comment (waynebruce0x) [inline projects/kicks/index.js:6]: can you use erc20:balanceOf() instead?

Comment (pavelGotbit) [reply projects/kicks/index.js:6]: In fact the calculation with ERC20::balanceOf will be incorrect due to the fact that the whole reward amount stays on the contract too, until users claim it.  Which means that we have to separately count TVL by calling totalSupply (which counts only users` liquidity)

**Comment by llamatester** [comment]
The adapter at projects/kicks exports TVL: 
        
 
 
    bsc-staking               18.96 k
    staking                   18.96 k
    bsc                       0
    
    total                    0

**Comment by llamatester** [comment]
The adapter at projects/kicks exports TVL: 
        
 
 
    bsc-staking               16.48 k
    staking                   16.48 k
    bsc                       0
    
    total                    0

**Pull request merged**